pytorch/pytorch
# Add namedtensor build and test to the CI

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #20168 [namedtensor] Add test/test_namedtensor.py
* **#20163 Add namedtensor build and test to the CI**
* #20162 Add USE_NAMEDTENSOR compilation flag.

These always run on master. In addition, these will only run on pull
requests if they have namedtensor in their title.

Right now, the build + test are equivalent to the normal pytorch
build with USE_NAMEDTENSOR=1 + test, if they run.

Test Plan:
- Check that the new namedtensor build + test runs on this PR and exits with success through `circleci step halt`
- Inspect the generated config.yml

gh-metadata: pytorch pytorch 20163 gh/zou3519/33/head

If BUILD_ENVIRONMENT contains

Differential Revision: [D15278216](https://our.internmc.facebook.com/intern/diff/D15278216)
